01-ruby-fundamentals/methods.md

```diff
@@ -97,6 +97,107 @@ puts result
 
 **Methods can often be referred to as _messages_. When we send a message, we expect a reply.**
 
+### Extended Example: Love Note
+
+To demonstrate how methods are useful, let's write a program that prints out a short love note to the console.
+
+#### The Heart
+
+We'll start with the most essential part of any love note: a heart.
+
+```ruby
+# love_note.rb
+def print_heart
+  puts "   .:::.   .:::.   "
+  puts "  :::::::.:::::::  "
+  puts "  :::::::::::::::  "
+  puts "  ':::::::::::::'  "
+  puts "    ':::::::::'    "
+  puts "      ':::::'      "
+  puts "        ':'        "
+end
+```
+
+Before we go any further we should test this method to make sure it works. Save this method in a file called `love_note.rb`, then from the command line load it into `irb`:
+
+```bash
+$ irb -r ./love_note.rb
+```
+
+This will drop us into the familiar `irb` prompt, with the added benefit that everything defined in the file `./love_note.rb` is available for use. Note that the `./` indicates `irb` should look in the current directory for the file - if you leave it off, it will show an error.
+
+Now from `irb` we can call our `print_heart` method:
+
+```
+2.4.0 :001 > print_heart
+   .:::.   .:::.    
+  :::::::.:::::::   
+  :::::::::::::::   
+  ':::::::::::::'   
+    ':::::::::'     
+      ':::::'       
+        ':'         
+ => nil
+```
+
+#### The Note
+
+Next we'll make use of this method... by writing another method! The new method, `print_love_note`, will take two parameters, `sender_name` and `receiver_name`, and it will make use of our `print_heart` method.
+
+**Question:** What will the _signature_ of this method look like? What might it _return_?
+
+Add the following code to `love_note.rb`, below our previous `print_heart` method:
+
+```ruby
+# love_note.rb
+def print_love_note(sender_name, receiver_name)
```

Comment:
The example is cute though honestly is feels a little bit yucky to me. Not a big fan of the love note context. Could we make it something with smily face ascii art instead?